src/sentry/api/endpoints/organization_plugins.py

```diff
@@ -6,21 +6,41 @@
 from sentry.api.bases.organization import OrganizationEndpoint
 from sentry.api.serializers import serialize
 from sentry.api.serializers.models.organization_plugin import OrganizationPluginSerializer
-from sentry.models import Project
+from sentry.models import Project, ProjectOption
 
 
 class OrganizationPluginsEndpoint(OrganizationEndpoint):
     def get(self, request, organization):
-        _plugins = []
-
-        for project in Project.objects.filter(organization=organization):
-            for plugin in plugins.configurable_for_project(project, version=None):
-                _plugins.append(
-                    serialize(
-                        plugin,
-                        request.user,
-                        OrganizationPluginSerializer(project),
-                    )
+        # Just load all Plugins once.
+        all_plugins = dict([
+            (p.slug, p) for p in plugins.all()
+        ])
+
+        if 'plugins' in request.GET:
+            desired_plugins = request.GET['plugins'].split()
+        else:
+            desired_plugins = all_plugins.keys()
+
+        projects = Project.objects.filter(
+            organization=organization,
+        )
+
+        # Each tuple represents an enabled Plugin (of only the ones we care
+        # about) and its corresponding Project.
+        enabled_plugins = ProjectOption.objects.filter(
+            key__in=['%s:enabled' % slug for slug in desired_plugins],
+            project_id__in=projects,
+        ).values_list('key', 'project_id')
+
+        resources = []
+
+        for key, project_id in enabled_plugins:
+            resources.append(
+                serialize(
+                    all_plugins[key.split(':')[0]],
+                    request.user,
+                    OrganizationPluginSerializer(projects.get(id=project_id)),
```

Comment:
We need to remove this O(n) query.

Reply:
Could you collect the project_ids in a list and do a single query for all of them?

Reply:
Yup. Exactly.

Reply:
Yup, I was just worried about loading ~800 objects into memory at once (an account with that many projects is what caused problems here in the first place).